poetry/utils/env.py

```diff
@@ -679,19 +681,38 @@ def create_venv(
 
     @classmethod
     def build_venv(
-        cls, path, executable=None
-    ):  # type: (Union[Path,str], Optional[Union[str, Path]]) -> virtualenv.run.session.Session
+        cls,
+        path,
+        executable=None,
+        with_pip=False,
+        with_wheel=None,
+        with_setuptools=None,
+    ):  # type: (Union[Path,str], Optional[Union[str, Path]], bool, Optional[bool], Optional[bool]) -> virtualenv.run.session.Session
         if isinstance(executable, Path):
             executable = executable.resolve().as_posix()
-        return virtualenv.cli_run(
-            [
-                "--no-download",
-                "--no-periodic-update",
-                "--python",
-                executable or sys.executable,
-                str(path),
-            ]
-        )
+
+        opts = [
+            "--no-download",
+            "--no-periodic-update",
+            "--python",
+            executable or sys.executable,
+        ]
+
+        if not with_pip:
+            opts.append("--no-pip")
+        else:
+            if with_wheel is None:
```

Comment:
What is the rationale for this?

Reply:
Added a comment for this as we need to handle cases where pip is enabled and wheel has not been explicitly disabled.